contents/handbook/engineering/ai/implementation.md

```diff
@@ -0,0 +1,74 @@
+
+**Session summaries** is callable from PostHog AI and Deep research, and also has its own UI. Use it when you need to analyze many session recordings and extract patterns or issues.
+
+**Array** is a desktop product for single-engineer use. It's separate from PostHog AI because the workflow is different — you're not asking questions, you're letting an AI agent watch PostHog for problems and automatically fix them in your codebase. Think of it as an AI assistant that lives in your development environment.
+
+**MCP** is for users who prefer to work in third-party tools like Claude Code or VS Code. You get access to PostHog's data and can combine it with other MCP servers (like Hubspot or Zendesk). The trade-off is you don't get PostHog AI's polished UX or PostHog-specific training.
+
+## Implementation recommendations
+
+### For engineers adding AI features
+
+If your feature **reads or writes PostHog data**, build it into PostHog AI or have it hand off to PostHog AI after initiation. For example, if you're adding a "Fix with AI" button to debug SQL queries, that button should open PostHog AI with context about the query, so users can iteratively debug with AI assistance.
+
+If your feature **triggers code changes**, feed it as a signal into the Array product. You can also offer a "copy prompt" option for engineers who don't want to use Array — they can paste the AI-generated prompt into their own code editor.
+
+If your feature **doesn't fit either category**, use your judgment and consult with the PostHog AI team if you're unsure. We're still figuring out where some things fit.
+
+## Future directions
+
+### Third-party context integration
+
+We want to connect PostHog AI to third-party tools for additional context. Imagine PostHog AI analyzing data across PostHog, Slack messages, and Zendesk tickets to understand not just what users are doing, but what they're saying and reporting. This data could also generate signals for the Array product — if users are complaining about a bug in Slack and PostHog sees errors in the same area, that's a strong signal for Array to investigate and potentially fix it automatically.
+
+This is in the idea stage right now, but the PostHog AI team will likely start working on it soon.
+
+### Continuous instrumentation
+
+The Wizard's future evolution involves continuous instrumentation — watching your codebase and suggesting event tracking for new features, filling gaps in existing tracking, and standardizing event patterns. This could integrate with Array to automatically handle PostHog instrumentation when generating code.
+
+### Mode expansion
+
+As product teams identify needs, we'll continue adding specialized agent modes. The mode architecture is designed to scale — teams can create their own modes without touching the core agent infrastructure.
```

Comment:
@kappa90 I am assuming the word 'mode' on this page still makes sense, if we are talking about technical modes? Otherwise we should call these "features". 

If we need to rename this anywhere, could you have another readthrough? It's hard for me to assess when we should use the technical term, or the marketing term.